### DISCUSSION TITLE
Wix4: Creating MSI launch conditions

### DISCUSSION THREAD

**Discussion by MSLukeWest** [wixtoolset/issues#6304] [Questions] [answered]
I'm trying to add a "Condition" element, which in v3 was a child of the product element, but doing this in v4 under the package element using wix.exe yields the error:

error WIX0005: The Package element contains an unexpected child element 'Condition'

Wix.exe convert doesn't edit the element, and running with -n doesn't report any errors.

I've looked at the schema (https://github.com/wixtoolset/doc/blob/main/src/xsd4/wix.xsd) and saw it can be a child of a fragment, so I tried doing that instead but I get an error similar to above.  The schema also still lists it as a child of Product, which doesn't seem to be used anymore in v4.

Does anyone know how to create launch conditions in v4?

**Comment by rseanhall** [comment, marked as answer]
Wix.exe convert is supposed to work. The tests are here: https://github.com/wixtoolset/Converters/blob/7378de4301d4f8c2570b29268ca7affbdb5721c2/src/test/WixToolsetTest.Converters/ConditionFixture.cs#L179. That should show you the new way, looks like the xsd file is out of date.

**Reply by MSLukeWest** [reply]
Thanks @rseanhall , that shows me the new syntax:


`<Launch Condition="1=2" Message="Stop" />`

I'm guessing the reason convert didn't initially work for me was because my condition was slightly more complicated than what's in that test:

```
<Condition Message="This application requires .NET Framework 4.7.2. Please install the .NET Framework then run this installer again.">
    <![CDATA[Installed OR WIX_IS_NETFRAMEWORK_472_OR_LATER_INSTALLED]]>
</Condition>
```

**Reply by robmen** [reply]
If that is the case, it would be great if you contributed a `Skip`ped  test demonstrating the conversion failure.

**Reply by rseanhall** [reply]
The lazy way to show code is with backticks like this but without the spaces at the beginning of the lines:

    ```
    code
    ```

Also, it looks like convert isn't working because you were using the new `Package` element, which was converted from the `Product` element after the `Condition` element was moved into an attribute so we don't have code to handle that scenario.

**Reply by MSLukeWest** [reply]
Yes you're right, I first ran convert on a very simple .wxs file and then copied the condition in and ran convert again later on.  I can see how that wouldn't be handled.  Thanks for the help and for pointing me to those tests.
